docs/controller.md

```diff
@@ -0,0 +1,108 @@
+# Architecture Document for Controller Component
+
+## Description
+The Controller is the component giving meaning to the [pipeline](#structure) file given by the Monitor. It receives [pipelines](#pipeline), parse them into a set of [actions](#actions) and send these actions sequentially to the Scheduler, for each of these actions, the Scheduler **must** notify the Controller when a job has been scheduled and has been completed successfully or encountered an error, it can be done just by sending the logs from the Agent for instance. Thanks to these information, the Scheduler is able to provide information about a pipeline state to anyone (the Monitor or any other client).
+## Features
+- Users can send pipelines containing actions to execute. These actions are basically shell commands to execute. These pipelines are `yaml` files.
+- Users can track there actions by getting the logs from the Agent, the states of the job : `TODO`, `PENDING`, `DOING`, `COMPLETED`, `ERROR`. Refer to the sections [actions/states](#States).
+- The controller makes sure that each actions are executed in the right order (by design) and doesn't execute the next action if the previous one has failed.
+## Pipeline
+A pipeline is a set of actions that are executed in sequence. It is represented as a `yaml` file (please, refer to the [structure](#structure) section for the reference of each sections of this file). A pipeline fails if one of its actions fail. If none of its actions fail then the pipeline succeeds.
+### Structure
+#### `actions`
+A pipeline is made up of one or more `actions`, which run sequentially. These actions declare a set of commands to run which implies to also declare a support to run on (for now this environment will be a container).
+#### `actions.<action_id>`
```

Comment:
Is that an HTTP endpoint or a section in the pipeline definition?

Reply:
A section in the pipeline definition. Maybe it's better to rename this part into "Pipeline declaration reference"

Reply:
Done